[Chua Wen Quan] iP

cwq2326 opened the pull request:
# Ekud
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
*  easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here.](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

cwq2326 committed:
Level 1. Greet, Echo, Exit

cwq2326 committed:
Level 2. Add, List

cwq2326 committed:
Level 3. Mark/ Unmark as Done

cwq2326 committed:
Level 4. ToDos, Events, Deadlines

cwq2326 committed:
A-TextUiTesting: Automated Text UI Testing

cwq2326 committed:
Level 5. Handle Errors

cwq2326 committed:
Level-6: Delete

cwq2326 committed:
A-Enums: Use Enums

cwq2326 committed:
Updated task bot welcome message and responses to bot commands

cwq2326 committed:
Updated input.txt, EXPECTED.TXT

cwq2326 committed:
Categorised bot commands into those needing and not needing description. Added a "commands" command for the bot to generate list of bot commands for the user.

cwq2326 committed:
Update coding standard for switch case statement

cwq2326 committed:
Remove wildcard imports

cwq2326 committed:
Implement Level-7: Save

cwq2326 committed:
Implement level 8. Dates and Times

cwq2326 committed:
Merge branch 'branch-Level-7'

cwq2326 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Event.java

cwq2326 committed:
Update TaskFunctions.class to DukeFunctions.class, change input of DukeFunctions.addToList to type LocalDate

cwq2326 committed:
Implement A-MoreOOP: Use More OOP

cwq2326 committed:
Implement A-Packages: Organize into Packages

cwq2326 committed:
Implement A-JUnit: Add JUnit Tests

cwq2326 committed:
Implement A-JavaDoc: JavaDoc

cwq2326 committed:
Implement A-CodingStandard

cwq2326 committed:
Implement Level 9. Find

cwq2326 committed:
Merge branch 'branch-Level-9'

cwq2326 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/DukeCommands.java

cwq2326 committed:
Update imports for test files

cwq2326 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DukeCommands.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/UnMarkCommand.java
#	src/main/java/duke/task/Task.java

cwq2326 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

cwq2326 committed:
Implement A-Gradle: Use Gradle

cwq2326 committed:
Implement A-CheckStyle: Use CheckStyle

cwq2326 committed:
Fix some java code style violations

cwq2326 committed:
Implement Level-10: GUI

cwq2326 committed:
Merge branch 'branch-Level-10'

cwq2326 committed:
Remove redundant method in command package

cwq2326 committed:
Fix clear command bug

louisdavinlie commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+    private MessageUi ui;
+    private Storage storage;
+    private TaskList tasks;
+    private boolean f;
```
Consider changing the name of this Boolean variable to a clearer one, according to the Java coding standard (for example doesFileExist)

louisdavinlie reviewed (commented):
Overall, excellent code! I could hardly find any problem in your code. I like how you created a Command interface and specific classes that implements Command for each command. Great work!

tanchinkiat99 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    public String getFileLoadingMessage() {
```
Would it be better to add documentation to describe the functionality of this function? 😃

tanchinkiat99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,128 @@
+import duke.command.OutputCommand;
+import duke.command.UnmarkCommand;
+import duke.exception.DukeException;
+import duke.task.Task;
```
I like how you organised these import statements in an orderly manner. 😸

tanchinkiat99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,128 @@
+                } catch (DateTimeParseException err) {
+                    throw new DukeException("Please enter a valid date! [dd/mm/yyyy]");
+                }
+                // Check if the index of task specified for task exist
```
I like how you used comments to show the logic of your if/else statements. 😄

tanchinkiat99 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,54 @@
+ * Represents commands which find tasks in the task list. A FindCommand
+ * object corresponds a valid Ekud command, which can then be executed.
+ */
+
```
I think it would be better to standardize the amount of space between documentation and your code to improve readability. 👍

tanchinkiat99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,69 @@
+     *
+     * @return Task description format for file input.
+     */
+    public String taskDescriptionForFile() {
```
Shouldn't you return this.description? 😄

tanchinkiat99 reviewed (commented):
Looks good. Just a bit of standardising to improve readability.

cwq2326 committed:
Add assert keywords to methods that contain switch statement, and other
methods that does not include the else statement.

There are several methods that uses switch statament and assumes that
the default branch will not be invoked, or the if statment catches
the alternate assumption of the variable.

Adding the assert keywords will eliminate these
assumptions.

As a step towards eliminating assumptions of the program, let's include
the assert keyword to methods that assumes that a variable will be of
the correct value.

cwq2326 committed:
Update code quality of the program

There are several codes in the program that violates the Java coding
standard such as code repetition.

Rectifying these violations improves the code quality.

As a step towards having good code quality, let's write code that
adheres to the coding standards.

cwq2326 committed:
Merge pull request #1 from cwq2326/branch-A-Assertions

Add assert keywords to methods that contain switch statement, and other

cwq2326 committed:
Merge branch 'master' of https://github.com/cwq2326/ip

cwq2326 committed:
Merge branch 'master' into branch-A-CodeQuality

cwq2326 committed:
Merge pull request #2 from cwq2326/branch-A-CodeQuality

Update code quality of the program

cwq2326 committed:
Merge branch 'master' of https://github.com/cwq2326/ip

cwq2326 committed:
Improve code quality and implement streams in DukeCommand class

cwq2326 committed:
Add new command: Postpone command

cwq2326 committed:
Improve code to adhere to style standards and remove wild card
imports in Parser class

cwq2326 committed:
Improve GUI and implement SLAP in the Parser class.

cwq2326 committed:
Set theme jekyll-theme-architect

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Add README.md and Ui.png

cwq2326 committed:
Merge branch 'master' of https://github.com/cwq2326/ip

# Conflicts:
#	docs/README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update user guide, bot personality and Ui.png

cwq2326 committed:
Update README.md

cwq2326 committed:
Update UserGuide

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update Ui.png, code to follow Java coding standard, bot dialog box
color

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Update README.md

cwq2326 committed:
Merge branch 'master' of https://github.com/cwq2326/ip

# Conflicts:
#	docs/README.md

cwq2326 committed:
Update README.md

Update README.md

cwq2326 committed:
Update README.md

Add image to README.md

cwq2326 committed:
Update README.md

Add alternate step for quick start
